demonstrations/tutorial_qubit_tapering.py

```diff
@@ -0,0 +1,274 @@
+Identity operators acting on M qubits
+
+.. math:: P_i \in \pm \left \{ I, X, Y, Z \right \} ^ {\bigotimes M}.
+
+The main idea in the symmetry-based qubit tapering approach is to find a unitary operator :math:`U`
+that transforms :math:`H` to a new Hamiltonian :math:`H'` which has the same eigenvalues as
+:math:`H`
+
+.. math:: H' = U^{\dagger} H U = \sum_{i=1}^r c_i \mu_i,
+
+such that each :math:`\mu_i` term in the new Hamiltonian acts trivially, e.g., with an Identity
+or a Pauli operator, on a set of qubits. This allows tapering-off those qubits from the Hamiltonian.
+For instance, consider the following Hamiltonian
+
+.. math:: H' = Z_0 X_1 - X_1 + Y_0 X_1,
+
+where all terms in the Hamiltonian act on the second qubit with the :math:`X` operator. It is
+straightforward to show that each term in the Hamiltonian commutes with :math:`I_0 X_1` and the
+ground state eigenvector of :math:`H'` is also an eigenvector of :math:`I_0 X_1` with eigenvalues
+:math:`\pm 1`. We can also rewrite the Hamiltonian as
+
+.. math:: H' = (Z_0 I_1 - I_0 I_1 + Y_0 I_1) I_0 X_1
+
+which gives us
+
+.. math:: H'|\psi \rangle = \pm1 (Z_0 I_1 - I_0 I_1 + Y_0 I_1)|\psi \rangle,
+
+where :math:`|\psi \rangle` is an eigenvector of :math:`H'`. This means that the Hamiltonian
+:math:`H` can be simplified as
+
+.. math:: H_{tapered} = \pm1 (Z_0 - I_0 + Y_0).
+
+The tapered Hamiltonian :math:`H_{tapered}` has the eigenvalues :math:`[-2.41421, 0.41421]` and
+:math:`[2.41421, -0.41421]` for the :math:`+1` and :math:`-1` eigenvalues, respectively. The
+eigenvalues of the original Hamiltonian :math:`H` are
+:math:`[2.41421356, -2.41421356,  0.41421356, -0.41421356]`
+
+More generally, we can construct the unitary :math:`U` such that each :math:`\mu_i` term acts with a
+Pauli-X operator on a set of qubits
+:math:`\left \{ j \right \}, j \in \left \{ l, ..., k \right \}` where :math:`j` is the qubit label.
+This guarantees that each term of the transformed Hamiltonian commutes with each of the Pauli-X
+operators applied to the :math:`j` qubit:
+
+.. math:: [H', X^j] = 0,
+
+and the eigenvectors of the transformed Hamiltonian :math:`H'` are also eigenvectors of each of the
+:math:`X^{j}` operators. Then we can factor out all of the :math:`X^{j}` operators from the
+transformed Hamiltonian and replace them with their eigenvalues :math:`\pm 1`. This gives us a
+set of tapered Hamiltonians depending on which eigenvalue :math:`\pm 1` we chose for each of the
+:math:`X^{j}` operator. For  instance, in the case of two tapered qubits, we have four eigenvalue
+sectors: :math:`[+1, +1]`, :math:`[-1, +1]`, :math:`[+1, -1]`, :math:`[-1, -1]`. In these tapered
+Hamiltonians, the set of :math:`\left \{ j \right \}, j \in \left \{ l, ..., k \right \}` qubits
+are eliminated. For tapered molecular Hamiltonians, it is possible to determine the optimal sector
+of the eigenvalues that corresponds to the ground-state. This has been explained in more details in

```

Comment:
```suggestion
of the eigenvalues that corresponds to the ground state. This is explained in more details in
```
"ground-state" is a compound adjective, while ground state is a an adjective "ground" describing a noun "state"